### ISSUE TITLE
Fails to upload a record with _id as non-objectId to mongodb 

### ISSUE THREAD

**Issue by beeplin** [Microsoft/vscode-cosmosdb#534]
Repro steps:

1. insert a record with a non-objectId `_id` to a collection in mongo shell. It shows in cosmosdb as:

```json
{
  "_id": "HJxU4ap5f",
  "title": "test393",
}
```

2. in cosmosdb, open the record and modify it with another title:

```json
{
  "_id": "HJxU4ap5f",
  "title": "new",
}
```
3. save it in cosmosdb, causing the following error.

Action: cosmosDB.CosmosEditorManager.onDidSaveTextDocument
Error type: Error
Error Message: Argument passed in must be a single String of 12 bytes or a string of 24 hex characters

Version: 0.6.0
OS: win32

**Comment by PrashanthCorp** [comment]
Were you dealing with Mongo documents when this error occured? 

I have seen this error before with Mongo documents. If it pertains to mongo, it could be due to a combination of factors - 
1. 12 bytes can be represented with 24 hex characters, so they are equivalent. Mongo's id fields are required to be 12bytes/24hex. Mongo can generate these for you, [called the `_id` or ObjectID](https://docs.mongodb.com/manual/reference/method/ObjectId/)
2. Mongo's `_id` field, which can also be referenced by `id`. See [the code](https://github.com/mongodb/mongoid/blob/master/lib/mongoid/fields.rb#L47) and a [stack overflow thread](https://stackoverflow.com/questions/9694460/difference-between-id-and-id-fields-in-mongodb)

Let us know if this helps.

**Comment by StephenWeatherford** [comment]
@PrashanthCorp This looks like a bug in our code.  Why are we forcing the _id to an ObjectId in the update code [here](https://github.com/Microsoft/vscode-cosmosdb/blob/3f25dffa764c5ce7a6b189cf34940aff9e07e9ce/src/mongo/tree/MongoDocumentTreeItem.ts#L71)?  _id's are not required to be ObjectId's

**Comment by Danlock** [comment]
Related to this issue, nested objectId are not set as object ids properly, in the case of:

```
{
  "_id": "5aa0150e1430142dda49b551",
  "createdAt": "2018-03-07T16:36:30.436Z",
  "data": {
    "Quasi": "8590406b-67fa-4e08-819d-2860f5fdca79"
  },
  "name": "g6d-car",
  "sensors": [
    {
      "_id": "5adf3b43194dd500010542ea",
      "createdAt": "2018-04-24T14:12:19.693Z",
      "type": "camera"
    }
  ],
  "type": "car",
  "updatedAt": "2018-04-24T14:12:19.693Z"
}
```

If I make any changes the inner sensor ids will be overwritten with "5adf3b43194dd500010542ea" instead of ObjectId("5adf3b43194dd500010542ea").

I can make another issue if you want, but this seems relevant.

**Comment by StephenWeatherford** [comment]
@Danlock Great observation, thanks!  @PrashanthCorp I recommend you go ahead and get the top-level fix in, and handle subdocuments in a separate PR (which will chnage the current fixes), so make sure you get testcases in.  Seems like it would actually be similar to the way we'd need to handle find.

**Comment by PrashanthCorp** [comment]
Thanks for pointing it out @Danlock ! 
If you don't mind, I'm moving your comment to a separate issue so that we can track it better.

**Comment by PrashanthCorp** [comment]
Opening, It looks like we never addressed the root issue filed.

**Comment by PrashanthCorp** [comment]
Hi @beeplin , @Danlock , the merged fix was published today (version 0.8.0). Hope this fixes your issue. :-)
Let us know how the fix holds up with your usage.
